persist-cli/src/main/java/io/ballerina/persist/nodegenerator/SyntaxTreeGenerator.java

```diff
@@ -105,51 +103,57 @@ public static SyntaxTree createPesistToml(String name) {
-                                    replaceAll("\"", ""))) {
-                                NodeList<KeyValueNode> subNodeList = node.fields();
-                                for (KeyValueNode subMember : subNodeList) {
-                                    if (subMember.identifier().toSourceCode().trim().equals(relatedKey)) {
-                                        persistConfig.put(configKey,
-                                                subMember.value().toSourceCode().trim());
-                                        configExists = true;
-                                    }
-
-                                }
-                            }
-
-                        }
-                    }
-                    if (!configExists) {
-                        throw new BalException(
-                                String.format("Persist.toml configuration template %s is not found in Config.toml ",
-                                        persistConfig.get(configKey).replaceAll("\"", "")));
-                    }
+            HashMap<String, TableNode> configs = new HashMap<>();
+            for (DocumentMemberDeclarationNode member : nodeList) {
+                if (member instanceof TableNode) {
+                    TableNode node = (TableNode) member;
+                    String tableName = node.identifier().toSourceCode().trim();
+                    configs.put(tableName, node);
                 }
             }
+            return configs;
         } catch (IOException e) {
             throw new BalException("Error while reading configurations. ");
         }
     }
 
+    public static void  populateConfigurations(ArrayList<String> templatedEntry, HashMap<String, String>
+            persistConfigurations, HashMap<String, TableNode> configs) throws BalException {
+        for (String configKey : templatedEntry) {
+            boolean configExists = false;
+            String[] placeHolderValues = persistConfigurations.get(configKey).replaceAll("\"", "").replaceAll("}", "")
+                    .split("\\.");
+            String relatedKey =  placeHolderValues[placeHolderValues.length - 1];
+            String placeHolderTable = persistConfigurations.get(configKey)
+                    .replaceAll("\"\\$\\{", "").replaceAll("}\"", "").replaceAll("\\." + relatedKey, "");
+            if (configs.containsKey(placeHolderTable)) {
+                TableNode tableNode = configs.get(placeHolderTable);
+                NodeList<KeyValueNode> subNodeList = tableNode.fields();
+                for (KeyValueNode subMember : subNodeList) {
+                    if (subMember.identifier().toSourceCode().trim().equals(relatedKey)) {
+                        persistConfigurations.put(configKey,
+                                subMember.value().toSourceCode().trim());
+                        configExists = true;
+                    }
+                }
+            }
+            if (!configExists) {
```

Comment:
Rather than giving error for every entry separately, can we add this to a list and print all.similarly to what we expect in Configurations not found in Config.toml

Reply:
updated in [df30e70](https://github.com/ballerina-platform/persist-tools/pull/65/commits/df30e70bcabe76fa78dc959280352bb2f08d8678), currently we get the error as follows 
```
Persist.toml configuration template ${foo.tool_test_db_push_1.clients.database} is not found in Config.toml. 
Persist.toml configuration template ${foo.tool_test_db_push_1.clients.root} is not found in Config.toml.

```